libs/pages/application/src/lib/feature/page-settings-resources-feature/page-settings-resources-feature.tsx

```diff
@@ -0,0 +1,110 @@
+import { useEffect, useState } from 'react'
+import { FieldValues, FormProvider, useForm } from 'react-hook-form'
+import { useDispatch, useSelector } from 'react-redux'
+import { useParams } from 'react-router-dom'
+import { editApplication, selectApplicationById } from '@console/domains/application'
+import { MemorySizeEnum } from '@console/shared/enums'
+import { ApplicationEntity } from '@console/shared/interfaces'
+import { convertCpuToVCpu, convertMemory } from '@console/shared/utils'
+import { AppDispatch, RootState } from '@console/store/data'
+import PageSettingsResources from '../../ui/page-settings-resources/page-settings-resources'
+
+export const handleSubmit = (data: FieldValues, application: ApplicationEntity, memorySize: MemorySizeEnum) => {
+  const cloneApplication = Object.assign({}, application)
+  const currentMemory = Number(data['memory'])
+
+  cloneApplication.memory = memorySize === MemorySizeEnum.GB ? currentMemory * 1024 : currentMemory
+  cloneApplication.cpu = convertCpuToVCpu(data['cpu'][0], true)
+  cloneApplication.min_running_instances = data['instances'][0]
+  cloneApplication.max_running_instances = data['instances'][1]
+
+  return cloneApplication
+}
+
+export function PageSettingsResourcesFeature() {
+  const { applicationId = '' } = useParams()
+
+  const [loading, setLoading] = useState(false)
+  const dispatch = useDispatch<AppDispatch>()
+
+  const application = useSelector<RootState, ApplicationEntity | undefined>(
+    (state) => selectApplicationById(state, applicationId),
+    (a, b) =>
+      a?.memory === b?.memory &&
+      a?.cpu === b?.cpu &&
+      a?.min_running_instances === b?.min_running_instances &&
+      a?.max_running_instances === b?.max_running_instances &&
+      JSON.stringify(a?.instances) === JSON.stringify(b?.instances)
+  )
+
+  const methods = useForm({
+    mode: 'onChange',
+    defaultValues: {
+      memory: application?.memory,
+      cpu: [convertCpuToVCpu(application?.cpu)],
+      instances: [application?.min_running_instances || 0, application?.max_running_instances || 1],
+    },
+  })
+
+  const [memorySize, setMemorySize] = useState<MemorySizeEnum>(MemorySizeEnum.MB)
+
+  useEffect(() => {
+    methods.setValue('memory', application?.memory)
+    methods.setValue('cpu', [convertCpuToVCpu(application?.cpu)])
+    methods.setValue('instances', [application?.min_running_instances || 1, application?.max_running_instances || 1])
+  }, [
+    methods,
+    application?.memory,
+    application?.cpu,
+    application?.min_running_instances,
+    application?.max_running_instances,
+  ])
+
+  const handleChangeMemoryUnit = () => {
```

Comment:
The `handleChangeMemoryUnit` function is not tested and could be externalised and factorised in some way since we are gonna have other places in the project where we can have this mb/gb conversion. Do you think it's possible/interesting or is it too case specific here?

Reply:
I saw by the way that the real converting function are already in utils and tested